zulipterminal/ui_tools/boxes.py

```diff
@@ -96,17 +98,20 @@ def stream_box_view(self, stream_id: int, caption: str='', title: str='',
             key=primary_key_for_command('AUTOCOMPLETE'),
             key_reverse=primary_key_for_command('AUTOCOMPLETE_REVERSE')
         )
-        self.stream_write_box = ReadlineEdit(
-            caption="Stream:  ",
-            edit_text=caption
-        )
-        self.stream_write_box.enable_autocomplete(
-            func=self._stream_box_autocomplete,
-            key=primary_key_for_command('AUTOCOMPLETE'),
-            key_reverse=primary_key_for_command('AUTOCOMPLETE_REVERSE')
-        )
-        self.stream_write_box.set_completer_delims("")
-
+        if not is_in_edit:
+            self.stream_write_box = ReadlineEdit(
+                caption="Stream:  ",
+                edit_text=caption
+            )
+            self.stream_write_box.enable_autocomplete(
+                func=self._stream_box_autocomplete,
+                key=primary_key_for_command('AUTOCOMPLETE'),
+                key_reverse=primary_key_for_command('AUTOCOMPLETE_REVERSE')
+            )
+            self.stream_write_box.set_completer_delims("")
+        else:
+            self.stream_write_box = urwid.AttrMap(urwid.SelectableIcon(
+                                    "Stream: " + caption), None, 'selected')
```

Comment:
Is there any particular reason why you've used `SelectableIcon`?

Reply:
I missed this one, thanks for pointing this out.

Reply:
I think that's because the type of edit shows a pop-up with options to choose from. It might be better to have the stream non-selectable.